app/routes.py

```diff
@@ -1 +1,172 @@
+        send_slack_notification(task.title)  # Send Slack notification
+        return jsonify(task=task.to_dict()), 200
+    else:
+        return jsonify(details="Task not found"), 404
+
+
+
+@tasks_bp.route("/<int:task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    task = Task.query.get(task_id)
+    if task:
+        task.completed_at = None
+        db.session.commit()
+        return jsonify(task=task.to_dict()), 200
+    else:
+        return jsonify(details="Task not found"), 404
+    
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_data = request.get_json()
+
+    if "title" not in request_data:
+        return jsonify(details="Invalid data"), 400
+
+    new_goal = Goal(title=request_data["title"])
+    db.session.add(new_goal)
+    db.session.commit()
+    return jsonify(goal=new_goal.to_dict()), 201
+
+@goals_bp.route("", methods=["GET"])
+def get_goals():
+    goals = Goal.query.all()
+    goals_response = [goal.to_dict() for goal in goals]
+    return jsonify(goals_response), 200
+
+@goals_bp.route("/<int:goal_id>", methods=["GET"])
+def get_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal:
+        return jsonify(goal=goal.to_dict()), 200
+    else:
+        return jsonify(details="Goal not found"), 404
+
+@goals_bp.route("/<int:goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal:
+        request_data = request.get_json()
+        goal.title = request_data.get("title", goal.title)
+        db.session.commit()
+        return jsonify(goal=goal.to_dict()), 200
+    else:
+        return jsonify(details="Goal not found"), 404
+
+@goals_bp.route("/<int:goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal:
+        db.session.delete(goal)
+        db.session.commit()
+        return jsonify(details=f'Goal {goal_id} "{goal.title}" successfully deleted'), 200
+    else:
+        return jsonify(details="Goal not found"), 404
+
+@goals_bp.route("/<int:goal_id>/tasks", methods=["POST"])
+def post_task_ids_to_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if not goal:
+        return jsonify(details="Goal not found"), 404
+
+    request_data = request.get_json()
+
+    if "task_ids" not in request_data:
+        return jsonify(details="Invalid data"), 400
+
+    for task_id in request_data["task_ids"]:
+        task = Task.query.get(task_id)
+        if task:
+            task.goal_id = goal_id
+            db.session.commit()
+        else:
+            return jsonify(details=f"Task {task_id} not found"), 404
+
+    return jsonify(id=goal_id, task_ids=request_data["task_ids"]), 200
+
+@goals_bp.route("/<int:goal_id>/tasks", methods=["GET"])
+def get_tasks_for_specific_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if not goal:
+        return jsonify(details="Goal not found"), 404
+
+    tasks = Task.query.filter_by(goal_id=goal_id).all()
```

Comment:
This isn't necessary! The `tasks` attribute in the Goal class is a list of tasks that are tied to a specific goal. You can use that to find the tasks and call `to_dict()` on each.